certification/internal/bundle/bundle.go

```diff
@@ -51,75 +61,69 @@ func Validate(ctx context.Context, operatorSdk operatorSdk, imagePath string) (*
 			opts.OptionalValues = make(map[string]string)
-			opts.OptionalValues["k8s-version"] = "1.22"
+			opts.OptionalValues["k8s-version"] = k8sVer
 		}
 	}
 
 	return operatorSdk.BundleValidate(ctx, imagePath, opts)
 }
 
-func isTarget49OrGreater(ocpLabelIndex string) bool {
-	semVerOCPV1beta1Unsupported, _ := semver.ParseTolerant(ocpVerV1beta1Unsupported)
-	// the OCP range informed cannot allow carry on to OCP 4.9+
+func targetVersion(ocpLabelIndex string) (string, error) {
 	beginsEqual := strings.HasPrefix(ocpLabelIndex, "=")
 	// It means that the OCP label is =OCP version
 	if beginsEqual {
 		version := cleanStringToGetTheVersionToParse(strings.Split(ocpLabelIndex, "=")[1])
 		verParsed, err := semver.ParseTolerant(version)
 		if err != nil {
-			log.Errorf("unable to parse the value (%s) on (%s)", version, ocpLabelIndex)
-			return false
+			return "", fmt.Errorf("unable to parse the value (%s) on (%s): %v", version, ocpLabelIndex, err)
 		}
 
-		if verParsed.GE(semVerOCPV1beta1Unsupported) {
-			return true
-		}
-		return false
+		return fmt.Sprintf("%d.%d", verParsed.Major, verParsed.Minor), nil
 	}
+
 	indexRange := cleanStringToGetTheVersionToParse(ocpLabelIndex)
 	if len(indexRange) > 1 {
-		// Bare version
+		// Bare version, so send back latest released
 		if !strings.Contains(indexRange, "-") {
 			verParsed, err := semver.ParseTolerant(indexRange)
 			if err != nil {
-				log.Error("unable to parse the version")
-				return false
+				// The passed version is not valid. We don't care what it is,
+				// just that it's valid.
+				return "", fmt.Errorf("unable to parse the version: %v", err)
 			}
-			if verParsed.GE(semVerOCPV1beta1Unsupported) {
-				return true
+
+			// If the specified version is greater than latestReleased, we will accept that
+			latestReleasedParsed, _ := semver.ParseTolerant(latestReleasedVersion)
+			if verParsed.GT(latestReleasedParsed) {
+				return fmt.Sprintf("%d.%d", verParsed.Major, verParsed.Minor), nil
 			}
+
+			return latestReleasedVersion, nil
 		}
 
 		versions := strings.Split(indexRange, "-")
-		version := versions[0]
-		if len(versions) > 1 {
-			version = versions[1]
+		// This is a normal range of 1.0-2.0
+		if len(versions) > 1 && versions[1] != "" {
+			version := versions[1]
 			verParsed, err := semver.ParseTolerant(version)
 			if err != nil {
-				log.Error("unable to parse the version")
-				return false
-			}
-
-			if verParsed.GE(semVerOCPV1beta1Unsupported) {
-				return true
+				return "", fmt.Errorf("unable to parse the version: %v", err)
 			}
-			return false
-		}
-
-		verParsed, err := semver.ParseTolerant(version)
-		if err != nil {
-			log.Error("unable to parse the version")
-			return false
+			return fmt.Sprintf("%d.%d", verParsed.Major, verParsed.Minor), nil
 		}
 
-		if semVerOCPV1beta1Unsupported.GE(verParsed) {
-			return true
-		}
+		// This is an open-ended range: v1-. This is not valid.
+		// So, we just fall through to the default return.
```

Comment:
It would seem that we _know_ the reason why this hitting this part of the code is invalid, but falling through to the default return implies we don't know why. Should we just return with a useful error message, even if it's an extra return?

Reply:
Valid.